File: lesson_13/maps_java/maps_app/src/main/java/com/codedifferently/lesson13/Lesson13.java
package com.codedifferently.lesson13;

import java.util.HashMap;

public class Lesson13 {

  /**
   * Provide the solution to LeetCode 3146 here:
   * https://leetcode.com/problems/permutation-difference-between-two-strings
   */
  public int findPermutationDifference(String s, String t) {
    HashMap<Character, Integer> indexMap = new HashMap<>();
    for (int i = 0; i < s.length(); i++) {
      indexMap.put(s.charAt(i), i);
    }
    int difference = 0;
    for (int i = 0; i < t.length(); i++) {
      if (indexMap.containsKey(t.charAt(i))) {
        difference += Math.abs(indexMap.get(t.charAt(i)) - i);
      }
    }
    return difference;
  }
}
// Used ChatGPT to explain code.
 

File: lesson_13/maps_ts/src/lesson13.ts
/**
 * Provide the solution to LeetCode 3146 here:
 * https://leetcode.com/problems/permutation-difference-between-two-strings
 */
export function findPermutationDifference(s: string, t: string): number {
 const indexMap = new Map <string, number>();
  for (let i = 0; i < s.length; i++) {
      indexMap.set(s[i], i);
  
    }
    let difference = 0;
     for (let i = 0; i < t.length; i++) {
      const originalIndex = indexMap.get(t[i]);
      //used chat to help check if it's undefined
     if (originalIndex !== undefined) {
       difference += Math.abs(originalIndex - i);
     }

    }
    return difference;
    
}

console.log(findPermutationDifference('abc', 'bac')); 
console.log(findPermutationDifference('abcde', 'edbac')); 